src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -109,6 +125,73 @@ public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem
     public static AbsolutePath FromDirectoryAndFileName(string? directoryPath, string fullPath, IFileSystem? fileSystem = null)
         => new(directoryPath, fullPath, fileSystem ?? Paths.FileSystem.Shared);
 
+    /// <summary>
+    /// Returns true if the given directory is a root directory on the current
+    /// platform.
+    /// </summary>
+    /// <param name="directory"></param>
+    /// <returns></returns>
+    /// <exception cref="PlatformNotSupportedException">
+    /// The current platform is not supported.
+    /// </exception>
+    internal static bool IsRootDirectory(ReadOnlySpan<char> directory)
+    {
+        var rootLength = GetRootLength(directory);
+        return rootLength == directory.Length;
+    }
+
+    /// <summary>
+    /// Gets the length of the root of the path.
+    /// </summary>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    /// <exception cref="PlatformNotSupportedException">
+    /// The current platform is not supported.
+    /// </exception>
+    private static int GetRootLength(ReadOnlySpan<char> path)
```

Comment:
This method [as well as `IsValidDriveChar`] would benefit from `[MethodImpl(MethodImplOptions.AggressiveInlining)]`, as they have very short bodies. 

It seems the JIT doesn't want to inline them, presumably due to multiple uses; but with the attributes they are successfully inlined.

Reply:
It's probably not inlining the function due to the exception being thrown.